### ISSUE TITLE
Crash on loading of the Android AwesomeProject App

### ISSUE THREAD

**Issue by eladgel** [facebook/react-native#3150]
When loading the AwesomeProject app for the first time, I get the next Crash in the logs
```
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime: FATAL EXCEPTION: main
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime: Process: com.awesomeproject, PID: 4920
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime: android.view.WindowManager$BadTokenException: Unable to add window android.view.ViewRootImpl$W@c56491a -- permission denied for this window type
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at android.view.ViewRootImpl.setView(ViewRootImpl.java:591)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:310)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:85)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at android.app.Dialog.show(Dialog.java:319)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at com.facebook.react.devsupport.DevSupportManager.handleReloadJS(DevSupportManager.java:471)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at com.facebook.react.ReactInstanceManager.initializeReactContext(ReactInstanceManager.java:326)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at com.facebook.react.ReactInstanceManager.attachMeasuredRootView(ReactInstanceManager.java:265)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at com.facebook.react.ReactRootView$1.run(ReactRootView.java:105)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at android.os.Handler.handleCallback(Handler.java:739)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at android.os.Handler.dispatchMessage(Handler.java:95)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at android.os.Looper.loop(Looper.java:148)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at android.app.ActivityThread.main(ActivityThread.java:5417)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at java.lang.reflect.Method.invoke(Native Method)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
09-30 18:54:08.050 4920-4920/com.awesomeproject E/AndroidRuntime:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

Going through the code sends me to line 464 in the DevSupportManager.java file in the ReactNative jar
```
 ProgressDialog progressDialog = new ProgressDialog(mApplicationContext);
    progressDialog.setTitle(R.string.catalyst_jsload_title);
    progressDialog.setMessage(mApplicationContext.getString(
        mIsUsingJSProxy ? R.string.catalyst_remotedbg_message : R.string.catalyst_jsload_message));
    progressDialog.setIndeterminate(true);
    progressDialog.setCancelable(false);
    progressDialog.getWindow().setType(WindowManager.LayoutParams.TYPE_SYSTEM_ALERT);
    progressDialog.show();
```
I tried adding the android.permission.SYSTEM_ALERT_WINDOW permission - Still crashing.
Will update soon if I fix this

**Comment by satya164** [comment]
@eladgel Android version?

**Comment by eladgel** [comment]
6.0 MPA44I

**Comment by BBLN** [comment]
https://commonsware.com/blog/2015/08/17/random-musings-android-6p0-sdk.html

https://stackoverflow.com/questions/32061934/permission-from-manifest-doesnt-work-in-android-6

**Comment by eladgel** [comment]
The second link did the trick for now :+1:

**Comment by deju** [comment]
@BBLN This works for me.

**Comment by iamrutvik** [comment]
@eladgel where have you written this?

**Comment by shercoder** [comment]
Ok so I followed the instructions shared by @BBLN. However, the app still crashes the first time I open the app. It is weird, in the background it will open the overlay permissions screen and the app crash occurs. Has anyone else experienced this? How did you mitigate this issue?

**Comment by aditijoshi** [comment]
https://medium.com/react-native-development/fixing-problems-in-react-native-caused-by-new-permission-model-on-android-1e547f754b8.


This worked for me.

**Comment by shercoder** [comment]
This issue has been fixed in RN 0.43
